### PR TITLE
form validation 

### PR DESCRIPTION
**Pull request by dwantula**

### CODE REVIEW

File: src/screens/BeerOfTheDay/BeerOfTheDay.jsx [hunk 1]
```diff
@@ -1,53 +1,49 @@
...
 function BeerOfTheDayComponent() {
-
   const [beer, setBeer] = useState({
     name: '',
     abv: '',
     img: '',
     ibu: '',
-    isPlaceholder: true
+    description: '',
+    isPlaceholder: true,
```
Comment (mi-chal) [inline src/screens/BeerOfTheDay/BeerOfTheDay.jsx]: Do you need this?

File: src/screens/BeerOfTheDay/BeerOfTheDay.jsx [hunk 2]
```diff
@@ -1,53 +1,49 @@
...
-    const newBeer = { id, name, description, img, ibu, abv };
-    const newBeers = [...beers, newBeer];
-    saveItemInLocalStorage('beers', newBeers);
-  };
+    const beers = getItemFromLocalStorage('beers') || []
+    const { id, img, name, description, ibu, abv } = beer
+    const newBeer = { id, name, description, img, ibu, abv }
+    const newBeers = [...beers, newBeer]
+    saveItemInLocalStorage('beers', newBeers)
+  }
```
Comment (mi-chal) [inline src/screens/BeerOfTheDay/BeerOfTheDay.jsx]: Why did you remove all `;` ?

File: src/screens/BeerOfTheDay/BeerOfTheDay.jsx [hunk 3]
```diff
@@ -1,53 +1,49 @@
...
       <Button
         className="button-choose"
         onClick={getRandomBeer}
         text="Choose a beer"
       />
-      {
-        !beer.isPlaceholder && <div className="random-beer">
-          <FavouriteButton
-            onClick={addBeerToFavourites} />
+      {!beer.isPlaceholder && (
```
Comment (mi-chal) [inline src/screens/BeerOfTheDay/BeerOfTheDay.jsx]: Do you need this?

File: src/screens/SearchBeers/SearchBeers.jsx [hunk 4]
```diff
@@ -137,172 +138,170 @@ const maltOptions = [
...
-      beerName: '',
-      yeast: '',
-      hops: '',
-      malt: '',
-    };
-  };
+function SearchBeersComponent() {
+
+  const [beers, setBeers] = useState([])
+  const [param, setParam] = useState({
```
Comment (mi-chal) [inline src/screens/SearchBeers/SearchBeers.jsx]: Should be named params

**Review by mi-chal — APPROVED** [review]

File: src/screens/BeerOfTheDay/BeerOfTheDay.jsx (hunk 1 above)
Comment (dwantula) [reply src/screens/BeerOfTheDay/BeerOfTheDay.jsx]: Fixed!

File: src/screens/BeerOfTheDay/BeerOfTheDay.jsx (hunk 3 above)
Comment (dwantula) [reply src/screens/BeerOfTheDay/BeerOfTheDay.jsx]: Fixed

File: src/screens/BeerOfTheDay/BeerOfTheDay.jsx (hunk 2 above)
Comment (dwantula) [reply src/screens/BeerOfTheDay/BeerOfTheDay.jsx]: Fixed!

File: src/screens/SearchBeers/SearchBeers.jsx (hunk 4 above)
Comment (dwantula) [reply src/screens/SearchBeers/SearchBeers.jsx]: Done

Comment (mi-chal) [inline src/screens/SearchBeers/SearchBeers.jsx]: Should be named params

File: src/screens/SearchBeers/SearchBeers.jsx [hunk 5]
```diff
@@ -137,172 +138,170 @@ const maltOptions = [
...
+      { prevParam, [name]: value })
+    );
   };
 
-  getBeers = async () => {
-    const { alcoholVolume, ibuRange, colorOfBeer, brewedBefore, food, yeast, hops, malt } = this.state;
-    const beers = await fetchBeers([alcoholVolume, ibuRange, colorOfBeer, brewedBefore, food, yeast, hops, malt]);
-    this.setState({ beers });
+  async function getBeers() {
+    const beers = await fetchBeers([param.alcoholVolume, param.ibuRange, param.colorOfBeer, param.brewedBefore, param.food, param.yeast, param.hops, param.malt]);
```
Comment (mi-chal) [inline src/screens/SearchBeers/SearchBeers.jsx]: Break it into separate lines

File: src/screens/SearchBeers/SearchBeers.jsx (hunk 4 above)
Comment (dwantula) [reply src/screens/SearchBeers/SearchBeers.jsx]: Done!

File: src/screens/SearchBeers/SearchBeers.jsx (hunk 5 above)
Comment (dwantula) [reply src/screens/SearchBeers/SearchBeers.jsx]: Fixed!

File: src/screens/BeerOfTheDay/BeerOfTheDay.jsx (hunk 1 above)
Comment (mi-chal) [inline src/screens/BeerOfTheDay/BeerOfTheDay.jsx]: Do you still need this?

File: src/screens/SearchBeers/SearchBeers.jsx [hunk 6]
```diff
@@ -137,172 +138,171 @@ const maltOptions = [
...
+    const beers = await fetchBeers([params.alcoholVolume, params.ibuRange, params.colorOfBeer, params.brewedBefore, params.food, params.yeast, params.hops, params.malt]);
+    setBeers(beers);
   };
 
-  getBeerByName = async () => {
-    const { beerName } = this.state
-    const beers = await fetchBeers([`beer_name=${beerName}`]);
-    this.setState({ beers })
-  }
+  async function getBeerByName() {
```
Comment (mi-chal) [inline src/screens/SearchBeers/SearchBeers.jsx]: Do you fetch single beer or more than 1?

File: src/screens/SearchBeers/SearchBeers.jsx [hunk 7]
```diff
@@ -137,172 +138,171 @@ const maltOptions = [
...
+  });
+
+  function handleChange(event) {
 
-  handleChange = event => {
     const { name, value } = event.target;
-    this.setState({
-      [name]: value
-    });
+    setParams(prevParam => (
```
Comment (mi-chal) [inline src/screens/SearchBeers/SearchBeers.jsx]: Should be prevParams

File: src/screens/SearchBeers/SearchBeers.jsx [hunk 8]
```diff
@@ -137,172 +138,171 @@ const maltOptions = [
...
 
-  addToFavorites = id => {
-    const beers = getItemFromLocalStorage('beers') || [];
-    const beer = this.state.beers.find(elem => elem.id === id);
-    const addBeer = [...beers, beer];
-    saveItemInLocalStorage('beers', addBeer);
+  function addToFavorites(id) {
+    const storedBeers = getItemFromLocalStorage('beers') || [];
+    const beer = beers.find(elem => elem.id === id);
+    const newBeer = [...storedBeers, beer];
```
Comment (mi-chal) [inline src/screens/SearchBeers/SearchBeers.jsx]: It is not a single beer

**Review by mi-chal — APPROVED** [review]

File: src/screens/BeerOfTheDay/BeerOfTheDay.jsx (hunk 1 above)
Comment (dwantula) [reply src/screens/BeerOfTheDay/BeerOfTheDay.jsx]: Fixed!

File: src/screens/SearchBeers/SearchBeers.jsx (hunk 6 above)
Comment (dwantula) [reply src/screens/SearchBeers/SearchBeers.jsx]: Fixed!

File: src/screens/SearchBeers/SearchBeers.jsx (hunk 7 above)
Comment (dwantula) [reply src/screens/SearchBeers/SearchBeers.jsx]: Done!

File: src/screens/SearchBeers/SearchBeers.jsx (hunk 8 above)
Comment (dwantula) [reply src/screens/SearchBeers/SearchBeers.jsx]: Done!
